diff --git a/container/image.bzl b/container/image.bzl
index f5b6d6b..98d9dc3 100644
--- a/container/image.bzl
+++ b/container/image.bzl
@@ -657,6 +657,8 @@ def _validate_command(name, argument, operating_system):
             return ["/bin/sh", "-c", argument]
     elif type(argument) == type([]):
         return argument
+    elif type(argument) == "select":
+        return argument
     elif argument:
         fail("The %s attribute must be a string or list, if specified." % name)
     else: