diff --git a/foreign_cc/private/framework.bzl b/foreign_cc/private/framework.bzl
index 2ba7236..10d1ae3 100644
--- a/foreign_cc/private/framework.bzl
+++ b/foreign_cc/private/framework.bzl
@@ -151,6 +151,10 @@ CC_EXTERNAL_RULE_ATTRIBUTES = {
         mandatory = False,
         default = "include",
     ),
+    "out_include_dirs": attr.string_list(
+        doc = "Optional name of the extra Output directories with header files",
+        mandatory = False,
+    ),
     "out_interface_libs": attr.string_list(
         doc = "Optional names of the resulting interface libraries.",
         mandatory = False,
@@ -417,6 +421,7 @@ def cc_external_rule_impl(ctx, attrs):
         bin_dir_name = attrs.out_bin_dir,
         lib_dir_name = attrs.out_lib_dir,
         include_dir_name = attrs.out_include_dir,
+        include_dir_names = attrs.out_include_dirs,
     )
     output_groups = _declare_output_groups(installdir_copy.file, outputs.out_binary_files)
     wrapped_files = [
@@ -614,6 +619,7 @@ def _check_file_name(var):
 _Outputs = provider(
     doc = "Provider to keep different kinds of the external build output files and directories",
     fields = dict(
+        out_include_dirs = "List of directory with header files",
         out_include_dir = "Directory with header files (relative to install directory)",
         out_binary_files = "Binary files, which will be created by the action",
         libraries = "Library files, which will be created by the action",
@@ -627,6 +633,7 @@ def _define_outputs(ctx, attrs, lib_name):
     attr_interface_libs = []
     attr_shared_libs = []
     attr_static_libs = []
+    attr_include_dirs = []
 
     # TODO: Until the the deprecated attributes are removed, we must
     # create a mutatable list so we can ensure they're being included
@@ -634,6 +641,7 @@ def _define_outputs(ctx, attrs, lib_name):
     attr_interface_libs.extend(getattr(attrs, "out_interface_libs", []))
     attr_shared_libs.extend(getattr(attrs, "out_shared_libs", []))
     attr_static_libs.extend(getattr(attrs, "out_static_libs", []))
+    attr_include_dirs.extend(getattr(attrs, "out_include_dirs", []))
 
     # TODO: These names are deprecated, remove
     if getattr(attrs, "binaries", []):
@@ -666,6 +674,10 @@ def _define_outputs(ctx, attrs, lib_name):
 
     _check_file_name(lib_name)
 
+    declared_include_dirs = []
+    for include_dir in attr_include_dirs:
+        declared_include_dirs.append(ctx.actions.declare_directory(lib_name + "/" + include_dir))
+
     out_include_dir = ctx.actions.declare_directory(lib_name + "/" + attrs.out_include_dir)
 
     out_binary_files = _declare_out(ctx, lib_name, attrs.out_bin_dir, attr_binaries_libs)
@@ -677,11 +689,13 @@ def _define_outputs(ctx, attrs, lib_name):
     )
 
     declared_outputs = [out_include_dir] + out_binary_files
+    declared_outputs += declared_include_dirs
     declared_outputs += libraries.static_libraries
     declared_outputs += libraries.shared_libraries + libraries.interface_libraries
 
     return _Outputs(
         out_include_dir = out_include_dir,
+        out_include_dirs = declared_include_dirs,
         out_binary_files = out_binary_files,
         libraries = libraries,
         declared_outputs = declared_outputs,
@@ -822,9 +836,16 @@ def _get_headers(compilation_info):
     )
 
 def _define_out_cc_info(ctx, attrs, inputs, outputs):
+    headers = [outputs.out_include_dir]
+    system_includes = [outputs.out_include_dir.path]
+
+    for include_dir in outputs.out_include_dirs:
+        headers.append(include_dir)
+        system_includes.append(include_dir.path)
+    
     compilation_info = cc_common.create_compilation_context(
-        headers = depset([outputs.out_include_dir]),
-        system_includes = depset([outputs.out_include_dir.path]),
+        headers = depset(headers),
+        system_includes = depset(system_includes),
         includes = depset([]),
         quote_includes = depset([]),
         defines = depset(attrs.defines),
diff --git a/foreign_cc/private/make_script.bzl b/foreign_cc/private/make_script.bzl
index c39661d..9a05845 100644
--- a/foreign_cc/private/make_script.bzl
+++ b/foreign_cc/private/make_script.bzl
@@ -97,6 +97,10 @@ def _define_deps_flags(deps, inputs):
 
                     dir_name = artifact.gen_dir.basename
                     include_dirs.append("-I$$EXT_BUILD_DEPS$$/{}/{}".format(dir_name, artifact.include_dir_name))
+
+                    for include_dir_name in artifact.include_dir_names:
+                        include_dirs.append("-I$$EXT_BUILD_DEPS$$/{}/{}".format(dir_name, include_dir_name))
+
                     lib_dirs.append("-L$$EXT_BUILD_DEPS$$/{}/{}".format(dir_name, artifact.lib_dir_name))
 
     return struct(
diff --git a/foreign_cc/providers.bzl b/foreign_cc/providers.bzl
index 695f9a0..73e54bd 100644
--- a/foreign_cc/providers.bzl
+++ b/foreign_cc/providers.bzl
@@ -19,6 +19,7 @@ Instances of ForeignCcArtifact are incapsulated in a depset ForeignCcDeps#artifa
         "bin_dir_name": "Bin directory, relative to install directory",
         "gen_dir": "Install directory",
         "include_dir_name": "Include directory, relative to install directory",
+        "include_dir_names": "Include directories, relative to install directory",
         "lib_dir_name": "Lib directory, relative to install directory",
     },
 )