diff --git a/BSW/amsr-vector-fs-socal/lib/socal/CMakeLists.txt b/BSW/amsr-vector-fs-socal/lib/socal/CMakeLists.txt
index 5757b49..f3fe9e5 100644
--- a/BSW/amsr-vector-fs-socal/lib/socal/CMakeLists.txt
+++ b/BSW/amsr-vector-fs-socal/lib/socal/CMakeLists.txt
@@ -63,10 +63,10 @@ target_link_libraries(${LIBRARY_NAME} PUBLIC
 )
 
 target_include_directories(${LIBRARY_NAME} PUBLIC
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+#  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>
-  "${amsr-vector-fs-libiostream_libstream_vector_stl_INCLUDE_DIRS}"
+#  "${amsr-vector-fs-libiostream_libstream_vector_stl_INCLUDE_DIRS}"
 )
 
 install(
