diff --git a/Makefile.am b/Makefile.am
index c02f6320..65e52ebb 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -591,8 +591,8 @@ liboconfig_la_SOURCES = \
 	src/liboconfig/oconfig.c \
 	src/liboconfig/oconfig.h \
 	src/liboconfig/aux_types.h \
-	$(srcdir)/src/liboconfig/scanner.l \
-	$(srcdir)/src/liboconfig/parser.y
+	src/liboconfig/scanner.l \
+	src/liboconfig/parser.y
 liboconfig_la_CPPFLAGS = -I$(srcdir)/src/liboconfig $(AM_CPPFLAGS)
 liboconfig_la_LDFLAGS = -avoid-version $(LEXLIB)
 
