diff --git a/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/src/amsr/thread/thread.cpp b/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/src/amsr/thread/thread.cpp
index 9dd405c2..145a0d47 100644
--- a/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/src/amsr/thread/thread.cpp
+++ b/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/src/amsr/thread/thread.cpp
@@ -16,6 +16,7 @@
  *
  *********************************************************************************************************************/
 
+#include <pthread.h>
 #include <algorithm>
 #include <memory>
 #include <utility>
diff --git a/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/src/amsr/thread/thread_error_domain.cpp b/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/src/amsr/thread/thread_error_domain.cpp
index ba91ed4a..32e6a452 100644
--- a/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/src/amsr/thread/thread_error_domain.cpp
+++ b/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/src/amsr/thread/thread_error_domain.cpp
@@ -16,6 +16,7 @@
  *
  *********************************************************************************************************************/
 
+#include <pthread.h>
 #include "amsr/thread/thread_error_domain.h"
 
 namespace amsr {
diff --git a/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/src/amsr/thread/thread_pool.cpp b/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/src/amsr/thread/thread_pool.cpp
index 09e981d3..659bd588 100644
--- a/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/src/amsr/thread/thread_pool.cpp
+++ b/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/src/amsr/thread/thread_pool.cpp
@@ -16,6 +16,7 @@
  *
  *********************************************************************************************************************/
 
+#include <pthread.h>
 #include "amsr/thread/common.h"
 #include "amsr/thread/thread_error_domain.h"
 #include "ara/core/result.h"
