diff --git a/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-per-libpersistency/lib/persistency-fs/CMakeLists.txt b/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-per-libpersistency/lib/persistency-fs/CMakeLists.txt
index e4de6ea9..9ccb9ddc 100644
--- a/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-per-libpersistency/lib/persistency-fs/CMakeLists.txt
+++ b/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-per-libpersistency/lib/persistency-fs/CMakeLists.txt
@@ -47,13 +47,19 @@ target_include_directories(${LIBRARY_NAME} PUBLIC
   "$<BUILD_INTERFACE:${VAC_INCLUDE_DIRS}>"
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
   "$<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>"
-  "$<BUILD_INTERFACE:$<TARGET_PROPERTY:${IO_INTEGRITY_STREAM_LIBRARIES},INTERFACE_INCLUDE_DIRECTORIES>>"
+  "$<BUILD_INTERFACE:${CRC_LIBCRC_INCLUDE_DIRS}>"
+  "$<BUILD_INTERFACE:${VAJSON_INCLUDE_DIRS}>"
   )
 
 target_link_libraries(${LIBRARY_NAME} PUBLIC
   ${VAC_LIBRARIES}
   persistency-common
   ${IO_INTEGRITY_STREAM_LIBRARIES}
+  ${VAC_LIBRARIES}
+  ${VAJSON_LIBRARIES}
+  ${CRC_LIBCRC_LIBRARIES}
+  ${amsr-vector-fs-libiostream_libstream_LIBRARIES}
   )
 
 export_library(${LIBRARY_NAME} FS_CMAKE_DIR OFF)
