diff --git a/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-libosabstraction/lib/CMakeLists.txt b/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-libosabstraction/lib/CMakeLists.txt
index 4d73a34f..da9cf0f3 100644
--- a/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-libosabstraction/lib/CMakeLists.txt
+++ b/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-libosabstraction/lib/CMakeLists.txt
@@ -114,9 +114,6 @@ if (vac_FOUND)
 endif ()
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
-  find_package(Threads REQUIRED)
-  message(STATUS "Package Threads found = ${Threads_FOUND}")
-  target_link_libraries(${LIBRARY_NAME} ${CMAKE_THREAD_LIBS_INIT})
   target_link_libraries(${LIBRARY_NAME} "rt")
 endif()
 
diff --git a/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/CMakeLists.txt b/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/CMakeLists.txt
index fe167dd1..42011cf4 100644
--- a/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/CMakeLists.txt
+++ b/mb_base_layer_adaptive_xavier/amsr_xavier/BSW/amsr-vector-fs-thread/lib/vathread/CMakeLists.txt
@@ -34,10 +34,9 @@ add_library(${LIBRARY_NAME} STATIC ${LIB_SRC})
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "QNX")
   # avoid using find_package Threads because generates invalid compiler flag for cc1plus (-pthread)
-else()
-  # use pthread library on systems w/ multiple thread libs
-  set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
-  find_package(Threads REQUIRED)
+elseif(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+  message(STATUS "Hardcoding dependency on pthread")
+  target_link_libraries(${LIBRARY_NAME} PRIVATE pthread)
 endif()
 
 find_package(vac REQUIRED)
