diff --git a/DaVinciConfigurator/Core/setupJreLinks.sh b/DaVinciConfigurator/Core/setupJreLinks.sh
index 117e56eb..dca84727 100755
--- a/DaVinciConfigurator/Core/setupJreLinks.sh
+++ b/DaVinciConfigurator/Core/setupJreLinks.sh
@@ -151,7 +151,6 @@ createJreLink() {
 # The original tarball from Oracle contains these symlinks and its planned to use and extract the tarball in future.
 #
 fixJreLinks() {
-  createMissingSymlink jcontrol "$SCRIPT_DIR/jre/bin/ControlPanel"
   createMissingSymlink ../libjsig.so "$SCRIPT_DIR/jre/lib/amd64/server/libjsig.so"
 }

